src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala

```diff
@@ -0,0 +1,56 @@
+package is.hail.expr.ir.functions
+
+import is.hail.annotations.{Region, StagedRegionValueBuilder}
+import is.hail.asm4s
+import is.hail.asm4s.{Code, MethodBuilder}
+import is.hail.expr.ir._
+import is.hail.expr.{Apply, ir, types}
+import is.hail.expr.types.{TBoolean, TBooleanOptional, TInterval, TStruct}
+import is.hail.utils._
+
+object IntervalFunctions extends RegistryFunctions {
+
+  def registerAll(): Unit = {
+
+    registerCode("Interval", Array(tv("T"), tv("T"), TBoolean(), TBoolean()), TInterval(tv("T")), isDet = true) {
+      (mb, args) =>
```

Comment:
Can you add a test suite for these like I did for CallFunctions?

Reply:
o7